File: oauth2-server/src/main/java/com/clouway/oauth2/ClientRequest.java
package com.clouway.oauth2;

import com.clouway.friendlyserve.Request;
import com.clouway.friendlyserve.Response;
import com.clouway.oauth2.client.ClientCredentials;
import com.clouway.oauth2.common.DateTime;

/**
 *
 * @author Miroslav Genov (miroslav.genov@clouway.com)
 */
public interface ClientRequest {

  Response handleAsOf(Request request, ClientCredentials credentials, DateTime instant);

}

File: oauth2-server/src/main/java/com/clouway/oauth2/ClientRequest.kt
package com.clouway.oauth2

import com.clouway.friendlyserve.Request
import com.clouway.friendlyserve.Response
import com.clouway.oauth2.client.ClientCredentials
import com.clouway.oauth2.common.DateTime

/**
 *
 * @author Miroslav Genov (miroslav.genov@clouway.com)
 */
interface ClientRequest {
	fun handleAsOf(request: Request, credentials: ClientCredentials, instant: DateTime): Response
}